lte/gateway/c/core/oai/tasks/amf/amf_config.c

```diff
@@ -662,7 +665,98 @@ int amf_config_parse_file(amf_config_t* config_pP) {
+      config_pP->amf_name = bfromcstr(astring);
+    }
+
+    // AMF_PLMN_SUPPORT SETTING
+    setting = config_setting_get_member(
+        setting_ngap, NGAP_CONFIG_AMF_PLMN_SUPPORT_LIST);
+    config_pP->plmn_support_count = 0;
+    if (setting != NULL) {
+      num = config_setting_length(setting);
+      OAILOG_DEBUG(LOG_AMF_APP, "Number of PLMN SUPPORT configured =%d\n", num);
+      AssertFatal(
+          num >= MIN_PLMN_SUPPORT,
+          "Not even one PLMN SUPPORT is configured, configure minimum one PLMN "
+          "LIST \n");
+      AssertFatal(
+          num <= MAX_PLMN_SUPPORT,
+          "Number of PLMN SUPPPORT configured:%d exceeds number of PLMN_SUPPORT"
+          ":%d \n",
+          num, MAX_PLMN_SUPPORT);
+
+      for (i = 0; i < num; i++) {
+        sub2setting = config_setting_get_elem(setting, i);
+
+        if (sub2setting != NULL) {
+          if ((config_setting_lookup_string(
+                  sub2setting, MME_CONFIG_STRING_MCC, &mcc))) {
+            AssertFatal(
+                strlen(mcc) == MAX_MCC_LENGTH,
+                "Bad MCC length (%ld), it must be %u digit ex: 001",
+                strlen(mcc), MAX_MCC_LENGTH);
+            char c[2]                                  = {mcc[0], 0};
+            config_pP->plmn_support[i].plmn.mcc_digit1 = (uint8_t) atoi(c);
+            c[0]                                       = mcc[1];
+            config_pP->plmn_support[i].plmn.mcc_digit2 = (uint8_t) atoi(c);
+            c[0]                                       = mcc[2];
+            config_pP->plmn_support[i].plmn.mcc_digit3 = (uint8_t) atoi(c);
+          }
+
+          if ((config_setting_lookup_string(
+                  sub2setting, MME_CONFIG_STRING_MNC, &mnc))) {
+            AssertFatal(
+                (strlen(mnc) == MIN_MNC_LENGTH) ||
+                    (strlen(mnc) == MAX_MNC_LENGTH),
+                "Bad MNC length (%ld), it must be %u or %u digit ex: 12 or 123",
+                strlen(mnc), MIN_MNC_LENGTH, MAX_MNC_LENGTH);
+            char c[2]                                  = {mnc[0], 0};
+            config_pP->plmn_support[i].plmn.mnc_digit1 = (uint8_t) atoi(c);
+            c[0]                                       = mnc[1];
+            config_pP->plmn_support[i].plmn.mnc_digit2 = (uint8_t) atoi(c);
+            if (3 == strlen(mnc)) {
+              c[0]                                       = mnc[2];
+              config_pP->plmn_support[i].plmn.mnc_digit3 = (uint8_t) atoi(c);
+            } else {
+              config_pP->plmn_support[i].plmn.mnc_digit3 = 0x0F;
+            }
+          }
+
+          if ((config_setting_lookup_string(
+                  sub2setting, NGAP_CONFIG_PLMN_SUPPORT_SST, &set_sst)) &&
+              strlen(set_sst)) {
+            config_pP->plmn_support[i].s_nssai.sst = (uint8_t) atoi(set_sst);
+          } else {
+            config_pP->plmn_support[i].s_nssai.sst =
+                NGAP_S_NSSAI_ST_DEFAULT_VALUE;
+          }
```

Comment:
we should not do this here. Please add it into config initialization and here it can be overwritten if specified in the config.

Reply:
+1

Reply:
Comments addressed.